client/layout/style.scss

```diff
@@ -134,6 +134,11 @@
 	overflow: hidden;
 }
 
+.layout.focus-sidebar:not(.is-section-post-editor) .layout__primary {
```

Comment:
Note that I still exclude editor, since it causes ~30px jump when we open the sidebar